predicators/envs/grid_row.py

```diff
@@ -325,27 +337,34 @@ def simulate(self, state: State, action: Action) -> State:
         ]
         assert len(door_cells) == 1
         # Apply ddoor if we're in same cell as door
+        # Can only open door, not close
         door_cell = door_cells[0]
         robot_cell = robot_cells[0]
-        if robot_cell == door_cell and not door_open and ddoor == 1.0:
-            next_state.set(self._door, "open", 1.0)
+        if robot_cell == door_cell and not (door_target - 0.1 \
+                            <= door_open <= door_target + 0.1 \
+    and door_target1 - 0.1 <= door_open1 <= door_target1 + 0.1):
+            new_door_level = np.clip(
+                state.get(self._door, "open") + ddoor, 0.0, 1.0)
+            next_state.set(self._door, "open", new_door_level)
+            new_door1_level = np.clip(
+                state.get(self._door, "open1") + ddoor1, 0.0, 1.0)
+            next_state.set(self._door, "open1", new_door1_level)
```

Comment:
Wait, so is it true that the robot could technically open the door with just one action (by setting the door level and door1 level correctly simultaneously?

Reply:
No, there isn't an action that handles both of the levels at once so it's forced to open the door through two actions (one TurnKey action and one MoveKey action)